src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -1143,9 +1143,13 @@ public boolean shouldBeInitialized(Class<?> c) {
     }
 
     /**
-     * Ensures the given class has been initialized. This is often
-     * needed in conjunction with obtaining the static field base of a
-     * class.
+     * Ensures the given class has been initialized (see JVMS-5.5 for details).
+     * This is often needed in conjunction with obtaining the static field base
+     * of a class.
+     *
+     * The call returns when either class {@code c} is fully initialized or
+     * class {@code c} is being initialized and the call is performed from
+     * the initializing thread.
```

Comment:
```suggestion
     * The call returns when either class {@code c} is fully initialized or
     * class {@code c} is being initialized and the call is performed from
     * the initializing thread. In the latter case a subsequent call to
     * {@link #shouldBeInitialized}, from the calling thread of this call,
     * will return {@code true}.
```

Reply:
Aren't "the calling thread of this call" and "the initializing thread" the same thread in the latter case?

Reply:
Agree. I dropped "from the calling thread of this call" part and incorporated the rest into the latest version.